recipes/default.rb

```diff
@@ -217,13 +216,37 @@
   mode 0750
   variables(
     {
-      :kafka_fqdn => kafka_fqdn,
       :mgm_fqdn => mgm_fqdn,
       :hopsworks_url => hopsworks_url
     }
   )
 end
 
+ruby_block 'copy-config-dir' do
+  block do
+    require 'fileutils'
+
+    if !node['onlinefs']['config_dir'].nil?
+      # Copy everything from the provided config_dir to etc overwriting any duplicates
+      FileUtils.cp_r(Dir["#{node['onlinefs']['config_dir']}/*"], node['onlinefs']['etc'])
+    end
+  end
```

Comment:
Instead of using the if statement here, use `only_if` in the ruby block. In this way it will show up in the logs if the step gets executed/skipped.